tests/unit/response_parsing/json/expected/elastictranscoder-delete-pipeline.json

```diff
@@ -1,5 +1,5 @@
 {
-    "ResponseMetadata": {},
+    "ResponseMetadata": {"RequestId": "1234"},
```

Comment:
As a (harmless or even desirable?) side effect, RestJSONParser now inherits the parsing RequestId ability from the adjusted JSONParser.

I actually mentioned it in the commit log as below.

    Besides, RestJSONParser will now grab x-amzn-requestid header
    and populate the ResponseMetadata.
    An old JSON response test case is updated accordingly.